test/cpp/test_tensor.cpp

```diff
@@ -370,6 +370,54 @@ TEST_F(TensorTest, TestAvgPool2DNonSquare) {
   }
 }
 
+TEST_F(TensorTest, TestBatchNorm1D) {
+  int num_features = 3;
+  at::Tensor input =
+      at::rand({2, num_features, 4}, at::TensorOptions(at::kFloat));
+  at::Tensor weight = at::rand({num_features}, at::TensorOptions(at::kFloat));
+  at::Tensor bias = at::rand({num_features}, at::TensorOptions(at::kFloat));
+  at::Tensor running_mean =
+      at::zeros({num_features}, at::TensorOptions(at::kFloat));
+  at::Tensor running_var =
+      at::ones({num_features}, at::TensorOptions(at::kFloat));
+  double momentum = 0.1;
+  double eps = 0.5;
+  at::Tensor undef;
+  for (bool training : {true, false}) {
+    for (bool undef_weight_bias : {true, false}) {
+      auto output = at::native_batch_norm(
```

Comment:
About the `auto`. This is not for this PR as I see there is some mess around this file, using both `auto` and `at::Tensor`.
The rule for using `auto` for us is if it is a templated type. So this does not qualify.
Will do a PR for it (or you can if you like).

Reply:
Btw I use `auto` here mainly to avoid spelling out `std::tuple<at::Tensor, at::Tensor, at::Tensor>`, does this qualify?  :P

Reply:
`auto` for `std::tuple<>` qualifies ;)

Reply:
I see, I'll push a commit to fix other usages in this test.

Reply:
Oh I see what you meant: this file has more `auto`s in other tests, I can send another PR for it after this.